prometheus/ha-mixin.libsonnet

```diff
@@ -35,67 +35,48 @@ local kausal = import 'ksonnet-util/kausal.libsonnet';
     },
   },
 
-  prometheus_zero+:: {
-    config+:: this.prometheus_config {
-      global+: {
-        external_labels+: {
-          __replica__: 'prometheus-0',
-        },
-      },
-    },
-    alerts+:: this.prometheusAlerts,
-    rules+:: this.prometheusRules,
-  },
-
-  prometheus_one+:: {
-    config+:: this.prometheus_config {
-      global+: {
-        external_labels+: {
-          __replica__: 'prometheus-1',
-        },
-      },
-    },
-    alerts+:: this.prometheusAlerts,
-    rules+:: this.prometheusRules,
-  },
-
   local configMap = k.core.v1.configMap,
-
-  prometheus_config_maps: [
-    configMap.new('%s-0-config' % _config.name) +
-    configMap.withData({
-      'prometheus.yml': k.util.manifestYaml(this.prometheus_zero.config),
-    }),
-    configMap.new('%s-0-alerts' % _config.name) +
-    configMap.withData({
-      'alerts.rules': k.util.manifestYaml(this.prometheus_zero.alerts),
-    }),
-    configMap.new('%s-0-recording' % _config.name) +
-    configMap.withData({
-      'recording.rules': k.util.manifestYaml(this.prometheus_zero.rules),
-    }),
-
-    configMap.new('%s-1-config' % _config.name) +
-    configMap.withData({
-      'prometheus.yml': k.util.manifestYaml(this.prometheus_one.config),
-    }),
-    configMap.new('%s-1-alerts' % _config.name) +
-    configMap.withData({
-      'alerts.rules': k.util.manifestYaml(this.prometheus_one.alerts),
-    }),
-    configMap.new('%s-1-recording' % _config.name) +
-    configMap.withData({
-      'recording.rules': k.util.manifestYaml(this.prometheus_one.rules),
-    }),
-  ],
+  prometheus_config_maps: std.mapWithIndex(
+    function(i, acc)
+      local name = _config.name + '-' + i;
+      local config =
+        this._config.prometheus_config {
+          prometheus+: {
+            global+: {
+              external_labels+: {
+                __replica__: name,
+              },
+            },
+          },
+        };
+      {
+        name: name,
+        base: configMap.new('%s-config' % name) +
+              configMap.withData({
+                'prometheus.yml': k.util.manifestYaml(config),
+              }),
+        alerts: configMap.new('%s-alerts' % name) +
+                configMap.withData({
+                  'alerts.rules': k.util.manifestYaml(this.prometheusAlerts),
+                }),
+        rules: configMap.new('%s-recording' % name) +
+               configMap.withData({
+                 'recording.rules': k.util.manifestYaml(this.prometheusRules),
+               }),
```

Comment:
We could take these out of the iteration and just provide the same rules/alerts to all replicas, but I didn't want to trigger a major diff.